File: frontend/src/app/controller/service/admin/stock/TacheAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {TacheDto} from 'src/app/controller/model/stock/Tache.model';
import {TacheCriteria} from 'src/app/controller/criteria/stock/TacheCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class TacheAdminService extends AbstractService<TacheDto, TacheCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/tache/');
    }

    public constrcutDto(): TacheDto {
        return new TacheDto();
    }

    public constrcutCriteria(): TacheCriteria {
        return new TacheCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/easystock/service/facade/admin/stock/TacheAdminService.java
package ma.zs.easystock.service.facade.admin.stock;

import java.util.List;
import ma.zs.easystock.bean.core.stock.Tache;
import ma.zs.easystock.dao.criteria.core.stock.TacheCriteria;
import ma.zs.easystock.zynerator.service.IService;



public interface TacheAdminService extends  IService<Tache,TacheCriteria>  {

    List<Tache> findByCollaborateurId(Long id);
    int deleteByCollaborateurId(Long id);
    long countByCollaborateurTelephone(String telephone);



}
